app/src/main/java/org/fossasia/openevent/activities/SettingsActivity.java

```diff
@@ -1,9 +1,43 @@
 package org.fossasia.openevent.activities;
 
+import android.content.Context;
+import android.os.Bundle;
+import android.preference.ListPreference;
+import android.preference.Preference;
+import android.preference.PreferenceActivity;
+import android.preference.PreferenceFragment;
+import android.preference.PreferenceScreen;
 import android.support.v7.app.ActionBarActivity;
+import android.util.Log;
+import android.view.MenuItem;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.fragments.SettingsFragment;
 
 /**
  * Created by manan on 21-05-2015.
  */
-public class SettingsActivity extends ActionBarActivity {
+public class SettingsActivity extends PreferenceActivity implements Preference.OnPreferenceChangeListener {
+    String pref;
+    Context mContext;
```

Comment:
Why do you need to cache the context? You are running inside Activity you can always access `this.getApplicationContext()`